src/client.rs

```diff
@@ -754,6 +772,255 @@ impl GeminiClient {
+        store.name = %store_name,
+        file.name = %request.file_name,
+    ))]
+    pub async fn import_file_to_search_store(
+        &self,
+        store_name: &str,
+        request: crate::file_search::ImportFileRequest,
+    ) -> Result<crate::file_search::Operation, Error> {
+        let url = self.build_url_with_suffix(&format!("{}:importFile", store_name))?;
+        self.post_json(url, &request).await
+    }
+
+    // Document operations
+
+    #[instrument(skip_all, fields(
+        store.name = %store_name,
+        document.id = %document_id,
+    ))]
+    pub async fn get_document(
+        &self,
+        store_name: &str,
+        document_id: &str,
+    ) -> Result<crate::file_search::Document, Error> {
+        let url =
+            self.build_url_with_suffix(&format!("{}/documents/{}", store_name, document_id))?;
+        self.get_json(url).await
+    }
+
+    #[instrument(skip_all, fields(
+        store.name = %store_name,
+        page.size = page_size,
+        page.token.present = page_token.is_some(),
+    ))]
+    pub async fn list_documents(
+        &self,
+        store_name: &str,
+        page_size: Option<u32>,
+        page_token: Option<&str>,
+    ) -> Result<crate::file_search::ListDocumentsResponse, Error> {
+        let mut url = self.build_url_with_suffix(&format!("{}/documents", store_name))?;
+        if let Some(size) = page_size {
+            url.query_pairs_mut()
+                .append_pair("pageSize", &size.to_string());
+        }
+        if let Some(token) = page_token {
+            url.query_pairs_mut().append_pair("pageToken", token);
+        }
+        self.get_json(url).await
+    }
+
+    #[instrument(skip_all, fields(
+        store.name = %store_name,
+        document.id = %document_id,
+        force,
+    ))]
+    pub async fn delete_document(
+        &self,
+        store_name: &str,
+        document_id: &str,
+        force: bool,
+    ) -> Result<(), Error> {
+        let mut url =
+            self.build_url_with_suffix(&format!("{}/documents/{}", store_name, document_id))?;
+        if force {
+            url.query_pairs_mut().append_pair("force", "true");
+        }
+        self.perform_request(|c| c.delete(url.clone()), async |_r| Ok(()))
+            .await
+    }
+
+    // Operation operations
+
+    #[instrument(skip_all, fields(operation.name = %name))]
+    pub async fn get_operation(&self, name: &str) -> Result<crate::file_search::Operation, Error> {
+        let url = self.build_url_with_suffix(name)?;
+        self.get_json(url).await
+    }
+
+    // Resumable upload helpers
+
+    #[instrument(skip(self, metadata))]
+    async fn initiate_resumable_upload<T: Serialize>(
+        &self,
+        path: &str,
+        total_bytes: usize,
+        mime_type: &Mime,
+        metadata: Option<&T>,
+    ) -> Result<String, Error> {
+        let url = self.build_url_with_suffix(path)?;
+
+        tracing::debug!("initiating resumable upload to {}", url);
```

Comment:
I think URL would be better to be the dedicated field (e.g. `url = %url`), not to be inlined into the trace message.